### PR TITLE
metrics,tidb: add StatementsPerTransaction and TransactionDuration

### PR DESCRIPTION
**Pull request by tiancaiamao**
StatementsPerTransaction is the write statement count for each transaction;
TransactionDuration is the execution duration for each transaction, including retry.

### CODE REVIEW

File: metrics/session.go [hunk 1]
```diff
@@ -78,6 +78,23 @@ var (
...
+			Subsystem: "server",
+			Name:      "statement_per_transaction",
+			Help:      "Write statements count in each transaction.",
+		}, []string{LblType})
+
+	TransactionDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "tidb",
+			Subsystem: "server",
+			Name:      "transaction_duration",
```
Comment (shenli) [inline metrics/session.go]: transaction_duration_seconds

File: session.go [hunk 2]
```diff
@@ -342,6 +342,11 @@ func (s *session) doCommitWithRetry(ctx goctx.Context) error {
 			err = s.retry(ctx, maxRetryCount)
 		}
 	}
+	if count := GetHistory(s).Count(); count > 1 {
```
Comment (shenli) [inline session.go]: Read-only statements are not in the history.

**Comment by zimulala** [comment]
LGTM

File: metrics/session.go [hunk 3]
```diff
@@ -78,6 +78,24 @@ var (
 			Name:      "session_restricted_sql_counter",
 			Help:      "Counter of internal restricted sql.",
 		})
+
+	StatementPerTransaction = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "tidb",
+			Subsystem: "server",
```
Comment (coocood) [inline metrics/session.go]: subsystem should be `session`.
Name better be `transaction_statement_num`.

File: metrics/session.go [hunk 4]
```diff
@@ -78,6 +78,24 @@ var (
...
+			Name:      "statement_per_transaction",
+			Help:      "Buckated histogram of statements count in each transaction.",
+			Buckets:   prometheus.ExponentialBuckets(1, 2, 12),
+		}, []string{LblType})
+
+	TransactionDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "tidb",
+			Subsystem: "server",
```
Comment (coocood) [inline metrics/session.go]: ditto

**Comment by shenli** [comment]
LGTM

**Pull request merged**